titus-api/src/main/java/com/netflix/titus/api/iam/model/IamRole.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  * Copyright 2019 Netflix, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.netflix.titus.api.iam.model;
+
+/**
+ * Interface for an IAM Role Policy.
+ */
+public interface IamRole {
```

Comment:
This class bundles data type (`IamRole`), with behavior (`canAssume`). I think canAssume should be part of `IamConnector`, or we should expose here a getter that provides all roles the given one can assume into.

We should also make the data type `IamRole` a regular JavaBean (no interface).

Reply:
Discussed offline. We'll move into the connector. Though moving away from subtyped IamRoles means another theoretical cloud connector's `IamRole` won't be explicit and deserializing the assume policy document in the connector may barf. Not ideal, but keeps the `IamRole` simpler.